test/patterns/if_return_if_detection/test_if_return_if.py

```diff
@@ -30,14 +30,14 @@ class TestCountIfReturn(TestCase):
     dir_path = Path(os.path.realpath(__file__)).parent
     countifreturn = CountIfReturn()
 
-    def test_no_return_inside(self):
-        lines = self.countifreturn.value(Path(self.dir_path, '2.java'))
-        self.assertEqual(lines, [])
-
     def test_2nd_level_inside(self):
         lines = self.countifreturn.value(Path(self.dir_path, '1.java'))
         self.assertEqual(lines, [6, 10])
 
+    def test_no_return_inside(self):
```

Comment:
+1

Reply:
nope, you are not right. To understant this idea in a proper way, you can read it: https://www.yegor256.com/2015/01/21/if-then-throw-else.html